lib/src/button/Button.tsx

```diff
@@ -118,43 +85,44 @@ const DxCButton = styled.div`
     props.margin && typeof props.margin === "object" && props.margin.bottom ? spaces[props.margin.bottom] : ""};
   margin-left: ${(props) =>
     props.margin && typeof props.margin === "object" && props.margin.left ? spaces[props.margin.left] : ""};
-
   display: inline-block;
   width: ${(props) => calculateWidth(props.margin, props.size)};
-  cursor: ${(props) => (props.disabled && "not-allowed") || "pointer"};
-
-  .MuiButtonBase-root {
-    padding-left: ${(props) => props.theme.paddingLeft};
-    padding-right: ${(props) => props.theme.paddingRight};
-    padding-top: ${(props) => props.theme.paddingTop};
-    padding-bottom: ${(props) => props.theme.paddingBottom};
-
-    .MuiButton-label {
-      display: flex;
-      align-items: center;
-    }
+`;
 
-    box-shadow: 0 0 0 2px transparent;
-    font-family: ${(props) => props.theme.fontFamily};
-    font-size: ${(props) => props.theme.fontSize};
-    font-weight: ${(props) => props.theme.fontWeight};
-    letter-spacing: ${(props) => props.theme.labelLetterSpacing};
-    min-width: ${(props) => (props.size === "small" && "calc(100% - 22px)") || "unset"};
-    width: 100%;
-    height: 40px;
-    transition: none !important;
+type ButtonProps = {
+  mode?: "primary" | "secondary" | "text";
+  size?: "small" | "medium" | "large" | "fillParent" | "fitContent";
+  backgroundType?: "dark" | "light";
+};
 
-    &:focus {
-      border-color: transparent;
-      box-shadow: 0 0 0 2px
-        ${(props) =>
-          props.backgroundType === "dark" ? props.theme.focusBorderColorOnDark : props.theme.focusBorderColor};
-    }
+const Button = styled.button<ButtonProps>`
+  padding-left: ${(props) => props.theme.paddingLeft};
+  padding-right: ${(props) => props.theme.paddingRight};
+  padding-top: ${(props) => props.theme.paddingTop};
+  padding-bottom: ${(props) => props.theme.paddingBottom};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  box-shadow: 0 0 0 2px transparent;
+  font-family: ${(props) => props.theme.fontFamily};
+  font-size: ${(props) => props.theme.fontSize};
+  font-weight: ${(props) => props.theme.fontWeight};
+  letter-spacing: ${(props) => props.theme.labelLetterSpacing};
+  min-width: ${(props) => (props.size === "small" && "calc(100% - 22px)") || "unset"};
+  width: 100%;
+  height: 40px;
+  cursor: pointer;
+  &:focus {
+    outline: none;
+    box-shadow: 0 0 0 2px
+      ${(props) =>
+        props.backgroundType === "dark" ? props.theme.focusBorderColorOnDark : props.theme.focusBorderColor};
+  }
 
-    ${(props) => {
-      const { mode, backgroundType } = props;
-      if (mode === "primary") {
-        return `
+  ${(props) => {
```

Comment:
What do you think about dividing this conditional into three components that inherit from Button?
I think that will make this a bit more readable.

Reply:
This it nos possible in Button so I made a refactor to avoid the 3 conditionals depending on the button mode and reduce code.